### PR TITLE
chore(deps): Upgrade to Rust 1.57

### PR DESCRIPTION
**Pull request by jszwedko**
Most of the changes were due to better detection of unused fields.

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: lib/vector-core/src/mapping/mod.rs [hunk 1]
```diff
@@ -143,6 +143,8 @@ impl Function for Noop {
 
 #[derive(Debug)]
 pub struct Mapping {
+    // this whole module needs to go away but I had trouble untangling it from what is _actually_ used by the legacy lookups
```
Comment (spencergilbert) [inline lib/vector-core/src/mapping/mod.rs:146]: 😞

**Review by spencergilbert — APPROVED** [review]

File: lib/vrl/tests/src/test.rs [hunk 2]
```diff
@@ -129,17 +129,21 @@ fn test_category(path: &Path) -> String {
     path.to_string_lossy()
         .strip_prefix("tests/")
         .expect("test")
-        .rsplitn(2, '/')
-        .nth(1)
-        .unwrap()
-        .to_owned()
+        .rsplit_once('/')
```
Comment (jszwedko) [inline lib/vrl/tests/src/test.rs:132]: https://rust-lang.github.io/rust-clippy/master/#manual_split_once

File: src/sinks/clickhouse.rs [hunk 3]
```diff
@@ -659,6 +659,7 @@ timestamp_format = "unix""#,
     }
 
     #[derive(Debug, Deserialize)]
+    #[allow(dead_code)] // deserialize all fields
```
Comment (jszwedko) [inline src/sinks/clickhouse.rs:662]: I could be persuaded that we should just leave out the fields that we don't care about deserializing. I just thought it's useful to see what the full payload is.

File: src/sources/exec/mod.rs [hunk 4]
```diff
@@ -501,7 +501,7 @@ fn spawn_reader_thread<R: 'static + AsyncRead + Unpin + std::marker::Send>(
     sender: Sender<((SmallVec<[Event; 1]>, usize), &'static str)>,
 ) {
     // Start the green background thread for collecting
-    Box::pin(tokio::spawn(async move {
+    let _ = Box::pin(tokio::spawn(async move {
```
Comment (jszwedko) [inline src/sources/exec/mod.rs:504]: Is there something better to do here? It was complaining that we weren't using the result of `Box::pin`.

**Comment by spencergilbert** [comment]
All of the changes appear reasonable/discussed in slack

**Review by bruceg — COMMENTED** [review]
This looks OK, but clippy appears to be erroring on more unused fields.

**Pull request merged**